src/OpenLoco/Industry.cpp

```diff
@@ -265,4 +266,57 @@ namespace OpenLoco
             }
         }
     }
+
+    // 0x004574F7
+    void Industry::updateProducedCargoStats()
+    {
+        const auto* industryObj = getObject();
+
+        for (auto cargoNum = 0; cargoNum < 2; ++cargoNum)
+        {
+            auto& indStatsStation = producedCargoStatsStation[cargoNum];
+            auto& indStatsRating = producedCargoStatsRating[cargoNum];
+            std::fill(std::begin(indStatsStation), std::end(indStatsStation), StationId::null);
+            const auto cargoType = industryObj->produced_cargo_type[cargoNum];
+            if (cargoType == 0xFF)
+            {
+                continue;
+            }
+
+            for (auto dword = 0; dword < 32; ++dword)
+            {
+                auto bits = var_E1[dword];
+                for (auto bit = Utility::bitScanForward(bits); bit != -1; bit = Utility::bitScanForward(bits))
+                {
+                    bits &= ~(1 << bit);
+                    const auto stationId = static_cast<StationId>((dword << 5) | bit);
+                    const auto* station = StationManager::get(stationId);
+                    if (station->empty())
+                    {
+                        continue;
+                    }
+
+                    const auto& cargoStats = station->cargoStats[cargoType];
+                    if (!(cargoStats.flags & (1 << 1)))
```

Comment:
any reason not to write `if (!(cargoStats.flags & 0x10))`

Reply:
i assume you mean 0x02. Thats how we write the flags before they are added anywhere. Its much easier to find in the codebase (1<<1) than it is to find 0x2 and if there were other flags its much easier to understand (1<<1)|(1<<0) compared to 0x3.

Reply:
yeah for some reason i was thinking in binary :| but no worries, was just wondering what convention was